include/examples/simple_io_behavior.h

```diff
@@ -0,0 +1,146 @@
+/**
+ * @file simple_io_behavior.h
```

Comment:
This behavior stuff is def something that you will want to share when you apply to jobs.

Reply:
Glad you think it's up to par!  I most certainly will.